src/test/java/fi/metatavu/metamind/test/functional/builder/impl/KnotTestBuilderResource.java

```diff
@@ -0,0 +1,182 @@
+package fi.metatavu.metamind.test.functional.builder.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import org.hibernate.search.hcore.impl.HibernateSearchIntegrator;
+import org.jboss.wsf.stack.cxf.interceptor.HandlerAuthInterceptor;
+import org.json.JSONException;
+
+import com.fasterxml.jackson.databind.deser.impl.ExternalTypeHandler.Builder;
+import com.oracle.truffle.js.nodes.cast.JSToPrimitiveNode.Hint;
+
+import feign.Body;
+import feign.FeignException;
+import fi.metatavu.metamind.ApiClient;
+import fi.metatavu.metamind.client.KnotsApi;
+import fi.metatavu.metamind.client.StoriesApi;
+import fi.metatavu.metamind.client.model.Knot;
+import fi.metatavu.metamind.client.model.KnotType;
+import fi.metatavu.metamind.client.model.Story;
+import fi.metatavu.metamind.client.model.TokenizerType;
+import fi.metatavu.metamind.test.functional.builder.AbstractTestBuilderResource;
+import fi.metatavu.metamind.test.functional.builder.TestBuilder;
+
+public class KnotTestBuilderResource extends AbstractTestBuilderResource<Knot, KnotsApi> {
+
+  public KnotTestBuilderResource(TestBuilder testBuilder, ApiClient apiClient) {
+    super(testBuilder, apiClient);
+  }
+
+  /**
+   * Creates a new Knot
+   * 
+   * @param type
```

Comment:
describe parameters (e.g. @param type knot type) and returns for all methods in this class